Druid/DRIRGenerator.class.st

```diff
@@ -42,11 +42,13 @@ DRIRGenerator >> addFrameReturn: aDRStackFrame [
 
 { #category : #visiting }
 DRIRGenerator >> addInstruction: instruction from: aNode [
-	 
+
 	instruction origin: thisContext sender stack.
 	self currentBasicBlock addInstruction: instruction.
 	instruction originAST: aNode.
 	self pushOperand: instruction.
+	instruction isStackInstruction ifTrue: [
+		self trackStackInstruction: instruction ].
```

Comment:
You're lazy xD You could have put this in the push and pop only, no?

Reply:
I did it first, but here is better, because some Generators on interpreting pop and push don't create the instructions (just affect the VMState or something like that).

.... and the code was uglier 😨